netsim/ansible/templates/initial/linux/vanilla.j2

```diff
@@ -46,24 +46,50 @@ ip link set {{ l.ifname }} mtu {{ l.mtu }}
 {% endif %}
 {% endfor %}
 #
-# Add routes to IPv4 address pools pointing to the first neighbor on the first link
+# Add static routes (usually IPv4 routes pointing to the first usable gateway)
 #
-# If you need anything better, use FRR instead of Linux and start routing (or use IPv6)
+{% set sr_list = routing.static|default([]) %}
+{% for pfx in sr_list|map(attribute='ipv4',default='')|unique if pfx %}
+{%   if loop.first %}
+set +e
+echo Removing existing IPv4 routes
+{%   endif %}
+ip route del {{ pfx }} 2>/dev/null
+{%   if loop.last %}
+set -e
+{%   endif %}
+{% endfor %}
 #
-{% for ifdata in interfaces|default([]) if ifdata.gateway is defined %}
-{%   for name,pool in pools.items()|default({}) %}
-{%     for af,pfx in pool.items() if af == 'ipv4' %}
-# {{ name }} prefix: {{ pfx }} local subnet: {{ ifdata.gateway.ipv4|ipaddr('subnet') }}
-{%       if name not in ['mgmt','router_id'] and pfx is string and
-            pfx != ifdata.gateway.ipv4|ipaddr('subnet') %}
+{% for pfx in sr_list|map(attribute='ipv6',default='')|unique if pfx %}
+{%   if loop.first %}
 set +e
+echo Removing existing IPv6 routes
+{%   endif %}
 ip route del {{ pfx }} 2>/dev/null
```

Comment:
If there are routes with multiple next hops, this will only delete the first such route. We may be better off simply flushing the entire route table at the start

Reply:
> If there are routes with multiple next hops, this will only delete the first such route.

Thank you. Fixed. A while loop makes the whole thing even easier ;)

> We may be better off simply flushing the entire route table at the start

That would also flush the static default route configured by containerlab plus whatever other custom stuff the user might have configured.